node/service/lib/registry.d.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import Device = require('./device');
+
+declare class Registry {
+    constructor(transport: Registry.Transport);
+    create(deviceInfo: { deviceId: string; }, done: Registry.ResponseCallback): void;
+    update(deviceInfo: { deviceId: string; }, done: Registry.ResponseCallback): void;
+    get(deviceId: string, done: Registry.ResponseCallback): void;
+    list(done: (err: Error, devices?: Device[], response?: any) => void): void;
+    delete(deviceId: string, done: Registry.ResponseCallback): void;
+    importDevicesFromBlob(inputBlobContainerUri: string, outputBlobContainerUri: string, done: Registry.JobCallback): void;
+    exportDevicesToBlob(outputBlobContainerUri: string, excludeKeys: boolean, done: Registry.JobCallback): void;
+    listJobs(done: (err: Error, jobsList?: string[]) => void): void;
+    getJob(jobId: string, done: Registry.JobCallback): void;
+    cancelJob(jobId: string, done: Registry.JobCallback): void;
+
+    static fromConnectionString(value: string, transportCtr?: Registry.TransportContructor): Registry;
+    static fromSharedAccessSignature(value: string): Registry;
+
+}
+
+declare namespace Registry {
+    interface TransportConfig {
+        host: string;
+        hubName: string;
+        sharedAccessSignature: string;
+    }
+    interface JobStatus {
+    }
```

Comment:
It's a server response, that might differ from transport to transport so I don't think you can expect common interface. Added this for better documentation.